### DISCUSSION TITLE
Why running KSP 1 requires `-language-version` to be `1.9`?

### DISCUSSION THREAD

**Discussion by wzieba** [ZacSweers/kotlin-compile-testing#309] [Q&A] [answered]
hi @ZacSweers 👋 Thank you for maintaining this fork. I have a question regarding PR https://github.com/ZacSweers/kotlin-compile-testing/pull/196 . You write:

> To use KSP, you must set `KotlinCompilation.languageVersion` to `1.9`.

Can you please share why exactly this is required?

**Comment by ZacSweers** [comment, marked as answer]
KSP1 only works in Kotlin 1.x. It is implemented on top of `AnalysisHandlerExtension`, which no longer exists in K2.

**Reply by ninniuz** [reply]
@ZacSweers by reading [this thread](https://slack-chats.kotlinlang.org/t/23220643/does-ksp-2-1-0-rc-1-0-26-support-ksp1-i-thought-kotlin-2-1-w) on Slack it seems to me that KSP1 should be still working with Kotlin 2.1.0 (that is actually what we are seeing in our projects, w/o KSP2 and with Kotlin 2.1.0).

But when using `kotlin-compile-testing` we don't get KSP to run unless we either set:
- `languageVersion = 1.9` or
- `configureKsp(useKsp2 = true)`

Do you know why this is happening?
We would like to execute our tests in the same env our annotation processing is performed in the app (more or less at least).
Thanks!

**Reply by ZacSweers** [reply]
Forcing language version to 1.9 forces the K1 compiler (even in K2). Please see the answer I gave above for why that's required https://github.com/ZacSweers/kotlin-compile-testing/discussions/309#discussioncomment-11549929. If you are using KSP in a project on kotlin 2.1.0 without KSP2, it is because KSP is forcing it down to languageVersion 1.9.
